plugins/woocommerce/includes/class-wc-form-handler.php

```diff
@@ -916,7 +916,7 @@ private static function add_to_cart_handler_variable( $product_id ) {
 		}
 
 		if ( false !== WC()->cart->add_to_cart( $product_id, $quantity, $variation_id, $variations ) ) {
-			wc_add_to_cart_message( array( $product_id => $quantity ), true );
+			wc_add_to_cart_message( array( $variation_id => $quantity ), true );
```

Comment:
this does not seem backward compatible, there might be custom code who would be expecting parent product here. can you also modify `wc_add_to_cart_message` such that in all filters and actions inside that method, we are still passing the parent object?

Reply:
I modified the function so that the parent object is still passed, but need to pass variation_id to some filters to generate the message.